### PR TITLE
Adapt write http plugin to new format

### LINKED ISSUE #544: write_http doesn't support Node
### Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 3.4.3
- Ruby: 1.9.3p484 (2013-11-22 revision 43786) [x86_64-linux]
- OS: Ubuntu 14.04
- Collectd version 5.5.2-1ubuntu0~ppa14.04.1
- Module version: 5.1.0

### How to reproduce (e.g Puppet code you use)
	class { 'collectd::plugin::write_http':
		urls => {
			'localhost:9999' => { 'format' => 'JSON', 'StoreRates' => false}
		}
	}

### What are you seeing
Failure when trying to start collectd:
 * Starting statistics collection and monitoring daemon collectd                                                                      write_http plugin: Legacy <URL> block found. Please use <Node> instead.

### What behaviour did you expect instead
I expected collectd to start.

### Output log
Error: Could not start Service[collectd]: Execution of '/etc/init.d/collectd start' returned 1:
Error: /Stage[main]/Collectd::Service/Service[collectd]/ensure: change from stopped to running failed: Could not start Service[collectd]: Execution of '/etc/init.d/collectd start' returned 1:

### Any additional information you'd like to impart
The new syntax is here: https://collectd.org/documentation/manpages/collectd.conf.5.shtml#plugin_write_http

### PR DESCRIPTION
**Pull request by kazeborja**
#### Pull Request (PR) description
Allows write http to define either legacy format (urls) or new format (nodes) as defined by the user.
Also adds tests for the write_http plugin that weren't there.

#### This Pull Request (PR) fixes the following issues
    Fixes #544

### CODE REVIEW

**Comment by kazeborja** [comment]
The failing check seems to be related to the CI flow, not much to the changes... Could someone confirm?

**Comment by kazeborja** [comment]
Anyway to retry the tests?

File: manifests/plugin/write_http.pp [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # https://collectd.org/wiki/index.php/Plugin:Write_HTTP
 class collectd::plugin::write_http (
-  $ensure    = 'present',
-  Hash $urls = {},
+  $ensure     = 'present',
+  Hash $nodes = {},
+  Hash $urls  = {}
```
Comment (bastelfreak) [inline manifests/plugin/write_http.pp]: can you enforce the structure of the hash? Examples are the last three here: https://puppet.com/docs/puppet/5.4/lang_data_hash.html#examples

File: templates/plugin/write_http.conf.erb [hunk 2]
```diff
@@ -1,6 +1,17 @@
 <Plugin "write_http">
-<% @urls.each do |name,values| -%>
+<% endpoints = {} -%>
```
[the remark this answers is not in the document]
Comment (kazeborja) [reply templates/plugin/write_http.conf.erb]: Sure thing, I can change it for an epp.

File: manifests/plugin/write_http.pp [hunk 3]
```diff
@@ -1,7 +1,8 @@
 # https://collectd.org/wiki/index.php/Plugin:Write_HTTP
 class collectd::plugin::write_http (
-  $ensure    = 'present',
-  Hash $urls = {},
+  $ensure     = 'present',
```
[the remark this answers is not in the document]
Comment (kazeborja) [reply manifests/plugin/write_http.pp]: Used enumeration as in some other plugins like CPU

File: manifests/plugin/write_http.pp (hunk 1 above)
Comment (kazeborja) [reply manifests/plugin/write_http.pp]: Added Hash structure definition

**Comment by kazeborja** [comment]
All discussions resolved, ready for a second look

File: manifests/plugin/write_http.pp [hunk 4]
```diff
@@ -1,13 +1,23 @@
 # https://collectd.org/wiki/index.php/Plugin:Write_HTTP
 class collectd::plugin::write_http (
-  $ensure    = 'present',
-  Hash $urls = {},
+  Enum['present', 'absent']          $ensure = 'present',
+  Hash[String, Hash[String, Scalar]] $nodes = {},
+  Hash[String, Hash[String, Scalar]] $urls  = {}
```
Comment (bastelfreak) [inline manifests/plugin/write_http.pp]: datatypes \o/

**Pull request merged**